mitiq/interface/mitiq_qibo/conversions.py

```diff
@@ -0,0 +1,531 @@
+    for gate in qibo_circuit.queue:
+        if isinstance(gate, gates.M):
+            continue
+        if gate.is_controlled_by:
+            raise UnsupportedQiboCircuitError(
+                "OpenQASM does not support multi-controlled gates."
+            )
+        if gate.name not in gates.QASM_GATES:
+            raise UnsupportedQiboCircuitError(
+                f"{gate.name} is not supported by OpenQASM."
+            )
+
+    # Decompose the circuit in known Cirq gates
+    qasm = decompose_qibo_circuit(qibo_circuit).to_qasm()
+
+    return cirq_from_qasm(qasm)
+
+
+def to_qibo(circuit: Circuit) -> QiboCircuit:
+    """Returns a QiboCircuit equivalent to the input Cirq circuit.
+
+    Args:
+        circuit: Cirq circuit to convert to a QiboCircuit.
+
+    Returns:
+        QiboCircuit object equivalent to the input Mitiq circuit.
+    """
+    
+    try:  
+        qasm = cirq_to_qasm(circuit)
+        nqubits, gate_list = _parse_qasm_modified(qasm)
+        
+    except ValueError: 
+        decomp_circuit = Circuit(decompose(circuit)) 
+        qasm = cirq_to_qasm(decomp_circuit)
+        nqubits, gate_list = _parse_qasm_modified(qasm)
+    
+    if nqubits < 1:
+        raise UnsupportedCirqCircuitError(
+            f"The number of qubits must be at least 1 but is {nqubits}."
+        )
+
+    qibo_circuit = QiboCircuit(nqubits)
+    for gate_name, qubits, params in gate_list:
+        gate = getattr(gates, gate_name)
+        if gate_name == "M":
+            qibo_circuit.add(gate(*qubits, register_name=params))
+        elif params is None:
+            qibo_circuit.add(gate(*qubits))
+        else:
+            # assume parametrized gate
+            qibo_circuit.add(gate(*qubits, *params))
+    
+    return qibo_circuit
+
+
+def read_args(args: str) ->  Generator[Tuple[str, int], None, None]:
+        _args = iter(re.split(r"[\[\],]", args))
+        for name in _args:
+            if name:
+                index = next(_args)
+                if not index.isdigit():
+                    raise_error(ValueError, "Invalid QASM qubit arguments:", args)
+                yield name, int(index)
+
+
+def _parse_qasm_modified(qasm_code: str) -> Tuple[int, List[Tuple[str, List[int], Union[List[float], str, None]]]]:
+    
+    """Extracts circuit information from QASM script.
+
+    Args:
+        qasm_code: String with the QASM code to parse.
+
+    Returns:
+        nqubits: The total number of qubits in the circuit.
+        gate_list: List that specifies the gates of the circuit.
+            Contains tuples of the form
+            (Qibo gate name, qubit IDs, optional additional parameter).
+            The additional parameter is the ``register_name`` for
+            measurement gates or ``theta`` for parametrized gates.
+    """
+     
+    lines = (line for line in "".join(
+        line for line in qasm_code.split("\n") if line and line[:2] != "//" and line[1:3] != "//"
+    ).split(";") if line and "//" not in line)
```

Comment:
I think this would be more readable if it was written as regular for loop. It's quite hard to understand what it's doing (and if all of the checks are necessary) as it stands.

Reply:
Also fixed by using latest version of qibo, QASM parsing removed.